### DISCUSSION TITLE
Changing the Browse Order, cont.

### DISCUSSION THREAD

**Discussion by eobrandon99** [CollectionBuilder/collectionbuilder.github.io#99] [General]
### Discussed in https://github.com/orgs/CollectionBuilder/discussions/72

<div type='discussions-op-text'>
<sup>Originally posted by **mldhistory** December 15, 2022</sup>
Hello!
I'm having trouble understanding the implementation of the config-browse.csv file. I would like to modify the sort order on the browse page of my collection so the default order shows the cards in the order that I have them entered in my .csv file. The current default is a random sort.  Also, how would I add additional sort-fields to the drop-down? The current list of sort fields in the dropdown (random,title,date) doesn't seem to correspond to the existing config-browse.csv file entries.
Any advice is much appreciated!
Thanks, Marcie</div>

IN REGARD TO THE ABOVE DISCUSSION
**Is there a way to have the Browse page sort to any other parameter (e.g. Title) by default?**

**Comment by evanwill** [comment]
@eobrandon99 yes, but it's not a configuration option so involves editing the template a bit. For example, to sort by Title by default:

In "_layouts/browse.html", you will find text inside a button element that looks like:

```
Sort by <span id="sortFilter">Random</span>
```

Replace the `Random` with `Title`.
Next, just below that look for the buttons that look like:

```
<button class="dropdown-item browse-sort-item active" data-filter="random">Random</button>
<button class="dropdown-item browse-sort-item" data-filter="title">Title</button>
```

Swap the order around like:

```
<button class="dropdown-item browse-sort-item active" data-filter="title">Title</button>
<button class="dropdown-item browse-sort-item" data-filter="random">Random</button>
```

Next in "_includes/js/browse-js.html", search for the lines that look like:

```
/* randomize items once at page load */
shuffle(items);
```

Comment out the shuffle and add a sorting function instead, so it looks like:

```
/* randomize items once at page load */
//shuffle(items);
/* sort by title at page load */
sorting(items,"title");
```

You could do the same with any other metadata field that you want to make default. 
(honestly, this is harder than it should be, we hope to re-write the browse js before too long, and might make this a configuration option or at least easier to do!)

**Reply by eobrandon99** [reply]
Hi! Thanks for your helpful reply. I just got around to incorporating this this and it worked great.

**Comment by evanwill** [comment]
@eobrandon99 I am converting this Issue into a Discussion so others can find it if they have the same idea.

**Comment by Digital-Grinnell** [comment]
Don't have time to properly address this just now, but I used this in https://yeager-collection.grinnell.edu/browse.html to set my default sort to "date".   If I had more time I'd be interested in re-working this so that sort order and default in Browse is configurable.  I considered adding a `sort-position` field to the end of `config-browse.csv` where the order and default could be set using integers, 0=first/default, 1=next pull-down option, etc.

**Comment by tprus-luc** [comment]
I have followed the steps listed in this thread to change the items to sorting by date. It is worth noting that this required changing all dates in the CSV data file to use YYYY-MM-DD format; dates using MM/DD/YYY format are sorted numerically: all January dates come first, then October, etc, and September last

**Reply by evanwill** [reply]
Yeah, for pretty much any of the "date" related stuff to work correct they need to be all in ISO yyyy-mm-dd format, https://collectionbuilder.github.io/cb-docs/docs/metadata/gh_metadata/#date

**Reply by tprus-luc** [reply]
Thanks for clarifying. The Metadata page says "(Dates in a mm/dd/yyyy format will also work)", and I am in the United States, so I have been using this format.
